pkg/scheduler/core/generic_scheduler.go

```diff
@@ -276,3 +283,130 @@ func (g *genericScheduler) divideReplicasByStaticWeight(clusters []*clusterv1alp
 	}
 	return targetClusters, nil
 }
+
+// TargetClustersList is a slice of TargetCluster that implements sort.Interface to sort by Value.
+type TargetClustersList []workv1alpha1.TargetCluster
+
+func (a TargetClustersList) Len() int           { return len(a) }
+func (a TargetClustersList) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a TargetClustersList) Less(i, j int) bool { return a[i].Replicas > a[j].Replicas }
+
+// divideReplicasAggregatedWithResource assigns a total number of replicas to the selected clusters aggregated according to the resource
+func (g *genericScheduler) divideReplicasAggregatedWithResource(clusters []*clusterv1alpha1.Cluster, object *workv1alpha1.ObjectReference) ([]workv1alpha1.TargetCluster, error) {
+	for _, value := range object.ReplicaResourceRequirements {
+		if value.Value() > 0 {
+			return g.divideReplicasAggregatedWithResourceRequirements(clusters, object)
+		}
+	}
+	return g.divideReplicasAggregatedWithoutResourceRequirements(clusters, object)
+}
+
+func (g *genericScheduler) divideReplicasAggregatedWithoutResourceRequirements(clusters []*clusterv1alpha1.Cluster,
+	object *workv1alpha1.ObjectReference) ([]workv1alpha1.TargetCluster, error) {
+	targetClusters := make([]workv1alpha1.TargetCluster, len(clusters))
+	for i, clusterInfo := range clusters {
+		targetClusters[i] = workv1alpha1.TargetCluster{Name: clusterInfo.Name, Replicas: 0}
+	}
+	targetClusters[0] = workv1alpha1.TargetCluster{Name: clusters[0].Name, Replicas: object.Replicas}
+	return targetClusters, nil
+}
+
+func (g *genericScheduler) divideReplicasAggregatedWithResourceRequirements(clusters []*clusterv1alpha1.Cluster,
+	object *workv1alpha1.ObjectReference) ([]workv1alpha1.TargetCluster, error) {
+	clusterAvailableReplicas := g.calAvailableReplicas(clusters, object.ReplicaResourceRequirements)
+	return g.divideReplicasAggregatedWithClusterReplicas(clusterAvailableReplicas, object.Replicas)
+}
+
+func (g *genericScheduler) calAvailableReplicas(clusters []*clusterv1alpha1.Cluster, replicaResourceRequirements corev1.ResourceList) []workv1alpha1.TargetCluster {
+	availableTargetClusters := make([]workv1alpha1.TargetCluster, len(clusters))
+	for i, cluster := range clusters {
+		maxReplicas := g.calClusterAvailableReplicas(cluster, replicaResourceRequirements)
+		availableTargetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name, Replicas: maxReplicas}
+	}
+	sort.Sort(TargetClustersList(availableTargetClusters))
+	return availableTargetClusters
+}
+
+// calClusterAvailableReplicas calculates how many replicas can be applied to the target cluster.
+func (g *genericScheduler) calClusterAvailableReplicas(cluster *clusterv1alpha1.Cluster, resourcePerReplicas corev1.ResourceList) int32 {
+	var maximumReplicas int64 = math.MaxInt32
+	resourceSummary := cluster.Status.ResourceSummary
+
+	for key, value := range resourcePerReplicas {
+		requestedQuantity := value.Value()
```

Comment:
@pigletfly 
Should we use `Value()` or `MilliValue()` here?

Reply:
@qianjun1993 
By the way, I think we need a unit test here.

Reply:
It should be MilliValue when the resource is 'CPU'